test/e2e/installer/helm.go

```diff
@@ -0,0 +1,121 @@
+// +build helm
+
+package installer
+
+import (
+	"fmt"
+	"time"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/assets/chart"
+	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
+)
+
+type clientAdapter struct {
+	*rest.Config
+}
+
+func (a clientAdapter) ToRESTConfig() (*rest.Config, error) {
+	if a.Config == nil {
+		return nil, fmt.Errorf("REST config is nil")
+	}
+	return a.Config, nil
+}
+
+func (a clientAdapter) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
+	cfg, err := a.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	dc, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return memory.NewMemCacheClient(dc), nil
+}
+
+func (a clientAdapter) ToRESTMapper() (meta.RESTMapper, error) {
+	dc, err := a.ToDiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(dc)
+	return restmapper.NewShortcutExpander(mapper, dc), nil
+}
+
+func (a clientAdapter) ToRawKubeConfigLoader() clientcmd.ClientConfig {
+	return clientcmd.ClientConfig(a)
+}
+
+func (a clientAdapter) RawConfig() (clientcmdapi.Config, error) {
+	return clientcmdapi.Config{}, fmt.Errorf("not supported")
+}
+
+func (a clientAdapter) ClientConfig() (*rest.Config, error) {
```

Comment:
ultra minor nit: function name could be a verb instead of a noun?

Reply:
This is an adapter to use a RESTConfig as [genericclioptions.RESTClientGetter](https://github.com/kubernetes/cli-runtime/blob/master/pkg/genericclioptions/config_flags.go#L71-L83), which Helm requires for initialization (check out where it's used below). I'll add a comment near the clientAdapter type definition to make this clear.